src/vmm/src/lib.rs

```diff
@@ -159,15 +203,25 @@ impl TryFrom<VMMConfig> for VMM {
         let vm_state = VmState {
             num_vcpus: config.vcpu_config.num,
         };
-        let vm = KvmVm::new(&kvm, vm_state, &guest_memory)?;
+
+        let wrapped_exit_handler = WrappedExitHandler(Arc::new(Mutex::new(VmmExitHandler {
+            exit_event: EventFd::new(libc::EFD_NONBLOCK).map_err(Error::ExitEvent)?,
+            keep_running: AtomicBool::new(true),
+        })));
+        let vm = KvmVm::new(&kvm, vm_state, &guest_memory, wrapped_exit_handler.clone())?;
+
+        let mut event_manager = EventManager::<Arc<Mutex<dyn MutEventSubscriber + Send>>>::new()
```

Comment:
It doesn't work because on the following line we are adding the `exit_handler` as a subscriber of `event_manager`, which makes Rust think that the type of subscriber is `event_manager::EventManager<Arc<Mutex<VmmExitHandler>>>`  instead of the expected `dyn MutEventSubscriber`.